[2015-08-27 12:30:41,327][ERROR, (HandlerDispatcher.java:128).run] 
[2015-08-27 12:32:40,094][ERROR, (HandlerDispatcher.java:128).run] io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1173)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1156)
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:612)
	at cpgame.demo.domain.Command.readString(Command.java:73)
	at cpgame.demo.domain.GameRequest.readString(GameRequest.java:99)
	at cpgame.demo.handler.InitHandler.execute(InitHandler.java:23)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 12:35:40,900][ERROR, (HandlerDispatcher.java:128).run] io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1173)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1156)
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:612)
	at cpgame.demo.domain.Command.readString(Command.java:73)
	at cpgame.demo.domain.GameRequest.readString(GameRequest.java:99)
	at cpgame.demo.handler.InitHandler.execute(InitHandler.java:23)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 12:36:18,918][ERROR, (HandlerDispatcher.java:128).run] io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1173)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1156)
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:612)
	at cpgame.demo.domain.Command.readString(Command.java:73)
	at cpgame.demo.domain.GameRequest.readString(GameRequest.java:99)
	at cpgame.demo.handler.InitHandler.execute(InitHandler.java:23)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 13:42:03,420][ERROR, (HandlerDispatcher.java:128).run] io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1173)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1156)
	at io.netty.buffer.AbstractByteBuf.readInt(AbstractByteBuf.java:612)
	at cpgame.demo.domain.Command.readString(Command.java:73)
	at cpgame.demo.domain.GameRequest.readString(GameRequest.java:99)
	at cpgame.demo.handler.InitHandler.execute(InitHandler.java:23)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 13:48:18,028][ERROR, (InitHandler.java:23).execute] Are you ok?
[2015-08-27 13:48:28,120][ERROR, (InitHandler.java:23).execute] Are you ok?
[2015-08-27 13:49:01,608][ERROR, (InitHandler.java:23).execute] Are you ok?
[2015-08-27 13:50:00,485][ERROR, (InitHandler.java:23).execute] Are you ok?
[2015-08-27 13:51:40,929][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 13:51:40,933][ERROR, (HandlerDispatcher.java:128).run] java.lang.ClassCastException: java.util.LinkedList cannot be cast to io.netty.buffer.ByteBuf
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:153)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 13:51:43,857][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 13:51:43,858][ERROR, (HandlerDispatcher.java:128).run] java.lang.ClassCastException: java.util.LinkedList cannot be cast to io.netty.buffer.ByteBuf
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:153)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 13:53:15,313][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 13:53:15,317][ERROR, (HandlerDispatcher.java:128).run] java.lang.ClassCastException: io.netty.handler.codec.http.websocketx.TextWebSocketFrame cannot be cast to io.netty.handler.codec.http.FullHttpRequest
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:148)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 13:59:09,836][ERROR, (InitHandler.java:23).execute] Are you ok?
[2015-08-27 13:59:35,547][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 13:59:35,552][WARN, (Slf4JLogger.java:151).warn] Failed to release a message.
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
	at io.netty.buffer.AbstractReferenceCountedByteBuf.release(AbstractReferenceCountedByteBuf.java:115)
	at io.netty.util.ReferenceCountUtil.release(ReferenceCountUtil.java:68)
	at io.netty.channel.ChannelOutboundBuffer.safeRelease(ChannelOutboundBuffer.java:509)
	at io.netty.channel.ChannelOutboundBuffer.remove(ChannelOutboundBuffer.java:268)
	at io.netty.channel.socket.nio.NioSocketChannel.doWrite(NioSocketChannel.java:266)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:682)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:271)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:651)
	at io.netty.channel.DefaultChannelPipeline$HeadHandler.flush(DefaultChannelPipeline.java:1038)
	at io.netty.channel.DefaultChannelHandlerContext.invokeFlush(DefaultChannelHandlerContext.java:675)
	at io.netty.channel.DefaultChannelHandlerContext.flush(DefaultChannelHandlerContext.java:656)
	at io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:115)
	at io.netty.channel.DefaultChannelHandlerContext.invokeFlush(DefaultChannelHandlerContext.java:675)
	at io.netty.channel.DefaultChannelHandlerContext.access$1600(DefaultChannelHandlerContext.java:29)
	at io.netty.channel.DefaultChannelHandlerContext$WriteTask.run(DefaultChannelHandlerContext.java:908)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Unknown Source)
[2015-08-27 14:06:09,251][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:06:23,581][WARN, (Slf4JLogger.java:151).warn] Failed to release a message.
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
	at io.netty.buffer.AbstractReferenceCountedByteBuf.release(AbstractReferenceCountedByteBuf.java:115)
	at io.netty.util.ReferenceCountUtil.release(ReferenceCountUtil.java:68)
	at io.netty.channel.ChannelOutboundBuffer.safeRelease(ChannelOutboundBuffer.java:509)
	at io.netty.channel.ChannelOutboundBuffer.remove(ChannelOutboundBuffer.java:268)
	at io.netty.channel.socket.nio.NioSocketChannel.doWrite(NioSocketChannel.java:266)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:682)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:271)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:651)
	at io.netty.channel.DefaultChannelPipeline$HeadHandler.flush(DefaultChannelPipeline.java:1038)
	at io.netty.channel.DefaultChannelHandlerContext.invokeFlush(DefaultChannelHandlerContext.java:675)
	at io.netty.channel.DefaultChannelHandlerContext.flush(DefaultChannelHandlerContext.java:656)
	at io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:115)
	at io.netty.channel.DefaultChannelHandlerContext.invokeFlush(DefaultChannelHandlerContext.java:675)
	at io.netty.channel.DefaultChannelHandlerContext.access$1600(DefaultChannelHandlerContext.java:29)
	at io.netty.channel.DefaultChannelHandlerContext$WriteTask.run(DefaultChannelHandlerContext.java:908)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Unknown Source)
[2015-08-27 14:11:09,319][ERROR, (HandlerDispatcher.java:132).run] java.lang.ClassCastException: io.netty.buffer.UnpooledHeapByteBuf cannot be cast to java.util.List
	at cpgame.demo.domain.GameResponse.write(GameResponse.java:77)
	at cpgame.demo.domain.GameResponse.<init>(GameResponse.java:37)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:130)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 14:12:02,713][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:12:02,718][ERROR, (HandlerDispatcher.java:132).run] java.lang.ClassCastException: java.util.LinkedList cannot be cast to io.netty.buffer.ByteBuf
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:160)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:130)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 14:12:54,859][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:21:51,899][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:13,408][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:14,448][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:14,938][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:15,338][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:15,598][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:15,838][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:18,949][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:20,339][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:37,331][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:22:52,441][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:23:05,799][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:31:40,958][ERROR, (HandlerDispatcher.java:132).run] java.lang.NullPointerException
	at cpgame.demo.domain.GameResponse.<init>(GameResponse.java:40)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:130)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

[2015-08-27 14:31:52,255][ERROR, (InitHandler.java:23).execute] are you ok?
[2015-08-27 14:32:16,807][ERROR, (InitHandler.java:23).execute] are you ok?
[2016-08-12 10:20:11,948][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:14:05,602][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:15:31,283][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:16:49,687][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:19:47,649][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:20:26,915][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:21:03,820][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:21:03,833][ERROR, (HandlerDispatcher.java:132).run] java.lang.NullPointerException
	at cpgame.demo.domain.GameResponse.<init>(GameResponse.java:39)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.handMessageQueue(HandlerDispatcher.java:140)
	at cpgame.demo.dispatcher.HandlerDispatcher$MessageWorker.run(HandlerDispatcher.java:130)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

[2016-08-12 14:21:53,715][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:25:52,469][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-12 14:26:53,975][ERROR, (InitHandler.java:23).execute] Are you ok?
[2016-08-17 15:09:34,358][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:81)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:09:44,293][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:81)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:09:54,292][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
[2016-08-17 15:09:54,296][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ReplayingDecoder.channelInactive(ReplayingDecoder.java:328)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelInactive(DefaultChannelHandlerContext.java:238)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelInactive(DefaultChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:767)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$5.run(AbstractChannel.java:560)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:81)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 9 more
[2016-08-17 15:14:01,930][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:14:11,891][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:14:21,892][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
[2016-08-17 15:14:21,895][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ReplayingDecoder.channelInactive(ReplayingDecoder.java:328)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelInactive(DefaultChannelHandlerContext.java:238)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelInactive(DefaultChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:767)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$5.run(AbstractChannel.java:560)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 9 more
[2016-08-17 15:22:15,875][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:22:25,773][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:139)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:338)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:324)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:126)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 10 more
[2016-08-17 15:22:35,774][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
[2016-08-17 15:22:35,777][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.IllegalStateException: Unknown command
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:417)
	at io.netty.handler.codec.ReplayingDecoder.channelInactive(ReplayingDecoder.java:328)
	at io.netty.channel.DefaultChannelHandlerContext.invokeChannelInactive(DefaultChannelHandlerContext.java:238)
	at io.netty.channel.DefaultChannelHandlerContext.fireChannelInactive(DefaultChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:767)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$5.run(AbstractChannel.java:560)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Unknown command
	at cpgame.demo.utils.RedisCommandDecoder.doSendCmdToHandler(RedisCommandDecoder.java:84)
	at cpgame.demo.utils.RedisCommandDecoder.decode(RedisCommandDecoder.java:29)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:362)
	... 9 more
[2016-08-18 15:35:25,861][WARN, (Slf4JLogger.java:151).warn] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: 远程主机强迫关闭了一个现有的连接。
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:384)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:871)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:208)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:118)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:485)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:452)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:346)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
	at java.lang.Thread.run(Thread.java:745)
